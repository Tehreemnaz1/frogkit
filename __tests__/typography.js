import { mount } from 'vue-test-utils'
import Vue from 'vue'
import Text from '@/components/typography'

const slot = { default: 'Lorem ipsum dolorem'}

describe('Text component', () => {

  it('should render component and match snapshot', () => {
    const wrapper = mount(Text, { slot })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render 50px text font size ', () => {
    const wrapper = mount(Text, { slot, propsData: { size: 'size-50'} })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render 38px text font size ', () => {
    const wrapper = mount(Text, { slot, propsData: { size: 'size-38'} })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render 30px text font size ', () => {
    const wrapper = mount(Text, { slot, propsData: { size: 'size-30'} })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render primary text color', () => {
    const wrapper = mount(Text, { slot, propsData: { color: 'primary'} })
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('should render light text color', () => {
    const wrapper = mount(Text, { slot, propsData: { color: 'light'} })
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('should render 700 font weight text', () => {
    const wrapper = mount(Text, { slot, propsData: { weight: 'weight-700' } })
    expect(wrapper.html()).toMatchSnapshot
  })

  it('should render 300 font weight text', () => {
    const wrapper = mount(Text, { slot, propsData: { weight: 'weight-300' } })
    expect(wrapper.html()).toMatchSnapshot
  })

  it('Text should emit click event when clicked', () => {
    const stub = jest.fn()
    const wrapper = mount(Text)
    wrapper.vm.$on('click', stub)
    wrapper.trigger('click')
    expect(stub).toBeCalled()
  })

})
