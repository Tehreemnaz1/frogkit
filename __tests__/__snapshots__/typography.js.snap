// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render component and match snapshot 1`] = `<div class="fk-txt"></div>`;

exports[`Text component should render lg text 1`] = `<div class="fk-txt fk-txt--lg"></div>`;

exports[`Text component should render light text color 1`] = `<div class="fk-txt fk-txt--light"></div>`;

exports[`Text component should render md text 1`] = `<div class="fk-txt fk-txt--md"></div>`;

exports[`Text component should render primary text color 1`] = `<div class="fk-txt fk-txt--primary"></div>`;

exports[`Text component should render xl text 1`] = `<div class="fk-txt fk-txt--xl"></div>`;
